cmake/OpenCogAtomTypes.cmake

```
@@ -241,8 +241,8 @@ FOREACH (LINE ${TYPE_SCRIPT_CONTENTS})
                 FILE(APPEND "${PYTHON_FILE}" "    return add_node(types.${TYPE_NAME}, node_name, tv)\n")
             ENDIF (ISNODE STREQUAL "NODE")
             IF (ISLINK STREQUAL "LINK")
-                FILE(APPEND "${PYTHON_FILE}" "def ${TYPE_NAME}(*args):\n")
-                FILE(APPEND "${PYTHON_FILE}" "    return add_link(types.${TYPE_NAME}, args)\n")
+                FILE(APPEND "${PYTHON_FILE}" "def ${TYPE_NAME}(*args, tv=None):\n")
+                FILE(APPEND "${PYTHON_FILE}" "    return add_link(types.${TYPE_NAME}, args, tv)\n")
```

I would explicitly mention that `tv` is a value for `tv` argument, otherwise there is a chance that it be used as part of outgoing:
```suggestion
                FILE(APPEND "${PYTHON_FILE}" "    return add_link(types.${TYPE_NAME}, args, tv=tv)\n")
```

---

Sorry, I don't understand what that means, could you expand @vsbogd?

---

Ah, OK, I get it, but I don't think there is any ambiguity because args is list (as far as I understand anyway), so it should consistently work, unless the API of `add_link` changes of course.

However your suggestion doesn't hurt so I might just as well apply it.

---

Ah agree, it is a problem only when arguments are unpacked.